src/iop/tonecurve.c

```diff
@@ -344,6 +347,19 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
                                          : d->table[ch_b][CLAMP((int)(b_in * 0x10000ul), 0, 0xffff)]);
         }
       }
+      else if(autoscale_ab == DT_S_SCALE_MANUAL_LCH)
+      {
+        const float pi = 3.14159f;
+        const float C_in = sqrtf(in[1]*in[1] + in[2]*in[2]) / 181.019f;
+        const float h_in = ((in[1]!=0.0f ? atanf(in[2]/in[1]) : (in[2]>0.0f ? 0.5f : -0.5f) * pi) + ((in[1] < 0.0f) ? pi : ((in[2] < 0.0f) ? 2.0f * pi : 0.0f))) / (2.0f * pi);
```

Comment:
there is an `atan2(y, x)` function doing just that (computing atan in the right quadrant)

Reply:
Great ! Done.